### ISSUE TITLE
Add An Abstraction Layer On Top Of Data Fetching

### ISSUE THREAD

**Issue by PengFei-Ethereum** [Uniswap/sybil-interface#40]
I will link to the discussion to that it's not forgotten : [https://github.com/Uniswap/sybil-interface/pull/31#issuecomment-749113304](url) . Essentially, we have noticed that governance structures among different projects such as Aave and Yearn make it difficult to properly integrate into Sybil. 

Mentioned in the discussion linked above is creating an abstraction layer on top of the data fetching, this would allow projects such as Aave and Yearn to maintain their current subgraphs while also being able to integrate into Sybil.

**Comment by ianlapham** [comment]
@PengFei-Ethereum 
Yep i think this is a good idea - the only issue that makes this more difficult is that Sybil allows users to delegate send actual transactions. 

Theres two different abstraction layers that would need to get built

1. provide a data abstraction layer for delegate data and proposal data 
2. provide transaction abstraction layer for contract interactions - this is hard because not all gov platform even enable the same functions 

I'd suggest that some compromise is to make a data abstraction layer and just hide voting unless its a direct fork of compond for now
